backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java

```diff
@@ -110,13 +110,15 @@
           .append(" ere." + ModelDBConstants.NAME + " = :experimentRunName ")
           .append(" AND ere." + ModelDBConstants.PROJECT_ID + " = :projectId ")
           .append(" AND ere." + ModelDBConstants.EXPERIMENT_ID + " = :experimentId ")
+          .append(" AND ere." + ModelDBConstants.DELETED + " = false ")
           .toString();
   private static final String CHECK_EXP_RUN_EXISTS_AT_UPDATE_HQL =
       new StringBuilder("Select count(*) From ExperimentRunEntity ere where ")
           .append(" ere." + ModelDBConstants.ID + " = :experimentRunId ")
+          .append(" AND ere." + ModelDBConstants.DELETED + " = false ")
           .toString();
   private static final String GET_EXP_RUN_BY_IDS_HQL =
-      "From ExperimentRunEntity exr where exr.id IN (:ids)";
+      "From ExperimentRunEntity exr where exr.id IN (:ids) AND exr.deleted = false ";
```

Comment:
Should we use ModelDBConstants.DELETED here too?

Reply:
updated with constant name